### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by msettles**
Adding a contributing doc, asking others to edit!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+HTStream was designed to quickly and easily add additional applications to the suite.
+
+## Pull Request Checklist
+
+Before sending your pull requests, make sure you followed this list.
+
+- Read [contributing guidelines](CONTRIBUTING.md).
+- Read [Code of Conduct](CODE_OF_CONDUCT.md).
+- Check if my changes are consistent with the [guidelines](https://github.com/ibest/HTStream/blob/master/CONTRIBUTING.md#general-guidelines-and-philosophy-for-contribution).
+- Run 'make check' and ensure all the Unit tests pass.
```
Comment (joe-angell) [inline CONTRIBUTING.md:12]: shouldn't this be `make test`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+HTStream was designed to quickly and easily add additional applications to the suite.
+
+## Pull Request Checklist
+
+Before sending your pull requests, make sure you followed this list.
+
+- Read [contributing guidelines](CONTRIBUTING.md).
+- Read [Code of Conduct](CODE_OF_CONDUCT.md).
+- Check if my changes are consistent with the [guidelines](https://github.com/ibest/HTStream/blob/master/CONTRIBUTING.md#general-guidelines-and-philosophy-for-contribution).
```
Comment (dstreett) [inline CONTRIBUTING.md:11]: I've read some of these that have a bullet something like `Open up an issue and ask about your feature. We would love to give guidance, save duplicate work if the feature is already under development, or save you some time if the functionality lives elsewhere.`

Pretty much encouraging people to talk about features being developed. But not a big deal, other than Joe's comment, looks great.

File: CONTRIBUTING.md (hunk 1 above)
Comment (msettles) [reply CONTRIBUTING.md]: fixed that

File: CONTRIBUTING.md (hunk 2 above)
Comment (msettles) [reply CONTRIBUTING.md]: there was language like that, down in contributing code, and not in guidelines make the language stronger thought

**Pull request merged**